applications/FluidDynamicsApplication/custom_processes/apply_mach_dependent_boundary_conditions.h

```diff
@@ -0,0 +1,314 @@
+
+            /// @brief Frees the Dof.
+            void Free(NodeType & rNode) const;
+
+        private:
+            const Variable<double> * mpVariable;
+            const double mValue; // Value to enforce
+            IntervalUtility mInterval;
+            
+            /* @brief Fixes the Dof managed by this class and sets the value
+             * to the one specified by mValue
+             */
+            static void FixDof(const BoundaryConditionUtility & rUtility, NodeType & rNode);
+
+            /** @brief Does nothing. Used to replace FixDof when t is outside
+             * time interval
+             */
+            static void DoNothing(const BoundaryConditionUtility & rUtility, NodeType & rNode);
+
+            /** @brief This object points to the proper (FixDof|DoNothing)
+             * function according to the time interval.
+             */
+            decltype(FixDof) * mEnforceInternal = & DoNothing;
+        };
+
+        /// Pointer definition of ApplyMachDependentBoundaryConditions
+        KRATOS_CLASS_POINTER_DEFINITION(ApplyMachDependentBoundaryConditions);
+
+        ///@}
+        ///@name Life Cycle
+        ///@{
+
+        /// Constructor
+        ApplyMachDependentBoundaryConditions(Model& rModel, Parameters Parameters);
+
+
+        ///@}
+        ///@name Operators
+        ///@{
+
+
+        ///@}
+        ///@name Operations
+        ///@{
+        
+        void ExecuteInitialize() override;
+
+        void ExecuteInitializeSolutionStep() override;
+
+        void ExecuteFinalizeSolutionStep() override;
+
+        const Parameters GetDefaultParameters() const override;
+
+        ///@}
+        ///@name Access
+        ///@{
+
+
+        ///@}
+        ///@name Inquiry
+        ///@{
+
+
+        ///@}
+        ///@name Input and output
+        ///@{
+
+        /// Turn back information as a string.
+        std::string Info() const override;
+
+        /// Print information about this object.
+        void PrintInfo(std::ostream& rOStream) const override;
+
+        /// Print object's data.
+        void PrintData(std::ostream& rOStream) const override;
+
+
+        ///@}
+        ///@name Friends
+        ///@{
+
+
+        ///@}
+
+        protected:
+        ///@name Protected static Member Variables
+        ///@{
+
+
+        ///@}
+        ///@name Protected member Variables
+        ///@{
+
+
+        ///@}
+        ///@name Protected Operators
+        ///@{
+
+
+        ///@}
+        ///@name Protected Operations
+        ///@{
+
+        ///@}
+        ///@name Protected  Access
+        ///@{
+
+
+        ///@}
+        ///@name Protected Inquiry
+        ///@{
+
+
+        ///@}
+        ///@name Protected LifeCycle
+        ///@{
+
+
+        ///@}
+
+        private:
+        ///@name Static Member Variables
+        ///@{
+
+
+        ///@}
+        ///@name Member Variables
+        ///@{
+
+        ModelPart * mpModelPart;
+        VectorVariable const * mpFlowDirectionVariable;
```

Comment:
I'd save a reference instead.

Reply:
For the modelpart it's simple enough, I can simply read the JSON before validating it.

But for the variable, it would entail calling KratosComponents::Get without validating the existance of the variable. This means that were the variable be misspelt, it would segfault without helping the user at all, at least when run not in debug mode. See https://github.com/KratosMultiphysics/Kratos/blob/c6b9da85068d55f8f6d9da01f9218f717d8bb1e0/kratos/includes/kratos_components.h#L96